File: marlin/src/main/java/com/hery/bigdata/conf/MarlinProperty.java
package com.hery.bigdata.conf;

import org.apache.commons.configuration.AbstractConfiguration;
import org.apache.commons.configuration.CombinedConfiguration;
import org.apache.commons.configuration.ConfigurationException;
import org.apache.commons.configuration.PropertiesConfiguration;
import org.apache.commons.configuration.tree.OverrideCombiner;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.File;
import java.net.URL;
import java.util.List;

/**
 * Created by houyongheng on 2019/7/23.
 * 配置文件维护
 */
public class MarlinProperty {
    private final static Logger logger = LoggerFactory.getLogger(MarlinProperty.class);

    private static CombinedConfiguration props = null;

    static {
        props = new CombinedConfiguration();
        props.setNodeCombiner(new OverrideCombiner());

        try {
            File file = new File("/data/bgservice/sparkapp/marlin/prod/marlin-prod.properties");
            final PropertiesConfiguration prodProps = new PropertiesConfiguration(file);
            logger.info("load prod config: {}", file.getPath());
            props.addConfiguration(prodProps);
        } catch (ConfigurationException e) {
            logger.warn("No prod property file");
        }

        try {
            URL defaultPropUrl = MarlinProperty.class.getResource("/marlin-local.properties");
            final PropertiesConfiguration defaultProps = new PropertiesConfiguration(defaultPropUrl);
            logger.info("load default config: {}", defaultPropUrl);
            props.addConfiguration(defaultProps);
        } catch (ConfigurationException e) {
            throw new RuntimeException("No default app property file");
        }
    }

    public String getString(String key) {
        return props.getString(key);
    }

    public Boolean getBoolean(String key){
        return props.getBoolean(key);
    }

    public int getInteger(String key) {
        return props.getInt(key);
    }


    public AbstractConfiguration getProperties() {
        return props;
    }
}

File: marlin/src/main/scala/com/weshare/marlin/utils/MarlinProperty.scala
package com.weshare.marlin.utils

import com.weshare.marlin.logging.WeshareLogging
import org.apache.commons.configuration.tree.OverrideCombiner
import org.apache.commons.configuration.{CombinedConfiguration, PropertiesConfiguration}

/**
  * Created by xuyi on 2018/1/31 16:42.
  */
object MarlinProperty extends WeshareLogging {

  val props: CombinedConfiguration = new CombinedConfiguration()
  props.setNodeCombiner(new OverrideCombiner)

  try {
    val prodPath = ""
    val prodProps: PropertiesConfiguration = new PropertiesConfiguration(prodPath)

    logger.info(s"Load prod config $prodPath")

    props.addConfiguration(prodProps)
  } catch {
    case ex: Exception =>
      logger.warn("No prod property file!")
  }

  try {
    val defaultPropUrl = MarlinProperty.getClass.getResource("/marlin-local.properties")
    val defaultProps: PropertiesConfiguration = new PropertiesConfiguration(defaultPropUrl)

    logger.info(s"load resource default config $defaultPropUrl")

    props.addConfiguration(defaultProps)
  } catch {
    case ex: Exception =>
      logger.warn("No resource default app property file!")
  }

  try {
    val defaultPropUrl = MarlinProperty.getClass.getResource("/marlin-local.properties")
    val defaultProps: PropertiesConfiguration = new PropertiesConfiguration(defaultPropUrl)

    logger.info(s"load resource default config $defaultPropUrl")

    props.addConfiguration(defaultProps)
  } catch {
    case ex: Exception =>
      logger.warn("No resource default app property file!")
  }

  def getString(key: String): String = {
    props.getString(key)
  }

  def getInteger(key: String): Int = {
    props.getInt(key)
  }
}
